scripts/asm_humidity_correction.py

```
@@ -0,0 +1,388 @@
+        # Generate the actual C++ code
+        cpp_code.append("// ============================================================================")
+        cpp_code.append("// ИСПРАВЛЕННЫЙ КОД ДЛЯ cropConfigs")
+        cpp_code.append("// ============================================================================")
+        cpp_code.append("")
+        
+        for crop_id, data in corrections.items():
+            if data["correction_needed"]:
+                cpp_code.append(f"// {crop_id.upper()} - {data['source']}")
+                cpp_code.append(f"// ASM: {data['asm_range']} → {data['recommended_asm']}%")
+                cpp_code.append(f"cropConfigs[\"{crop_id}\"] = CropConfig(")
+                cpp_code.append(f"    // temperature, humidity(ASM), EC, pH")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)